ui/lib/sync/addon/components/secrets/page/destinations.ts

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: BUSL-1.1
+ */
+
+import Component from '@glimmer/component';
+import { service } from '@ember/service';
+import { action } from '@ember/object';
+import { getOwner } from '@ember/application';
+import errorMessage from 'vault/utils/error-message';
+import { findDestination } from 'core/helpers/sync-destinations';
+
+import type SyncDestinationModel from 'vault/vault/models/sync/destination';
+import type RouterService from '@ember/routing/router-service';
+import type StoreService from 'vault/services/store';
+import type FlashMessageService from 'vault/services/flash-messages';
+import type { EngineOwner } from 'vault/vault/app-types';
+import type { SyncDestinationName, SyncDestinationType } from 'vault/vault/app-types';
+
+interface Args {
+  destinations: Array<SyncDestinationModel>;
+  nameFilter: SyncDestinationName;
+  typeFilter: SyncDestinationType;
+}
+
+export default class SyncSecretsDestinationsPageComponent extends Component<Args> {
+  @service declare readonly router: RouterService;
+  @service declare readonly store: StoreService;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  // typeFilter arg comes in as destination type but we need to pass the name of the type into the SearchSelect
+  get typeFilter() {
+    return findDestination(this.args.typeFilter)?.name;
+  }
+
+  get destinationNames() {
+    return this.args.destinations.map((destination) => ({ id: destination.name, name: destination.name }));
+  }
+
+  get destinationTypes() {
+    return this.args.destinations.reduce((types: Array<{ id: string; name: string }>, destination) => {
+      const { typeDisplayName } = destination;
+      const isUnique = !types.find((type) => type.name === typeDisplayName);
+      if (isUnique) {
+        types.push({ id: typeDisplayName, name: destination.type });
```

Comment:
I'm a little confused why destination.type is the `name` key here? I'd expect the id to be the `type`, and the display name to be `name`, switching the order of `objectKeys` would change the value rendered in the dropdown from the `id` to `name` 

But...if this works as is, no need to refactor! As we all know, Search Select is a pain 😭 so leaving as-is makes sense to me!